Log management system (#5801)
fastify/fastify: docs/Reference/Server.md

> Server.md
> 
> 

Reviewer: ```suggestion
```

What is this?

Author: Used as a notation.

> <h1 align="center">Fastify</h1>

Author: Absolutely, I will make sure to revert this change, my apologies for the oversight.

  - [`onProtoPoisoning`](#onprotopoisoning)
  - [`onConstructorPoisoning`](#onconstructorpoisoning)
  - [`logger`](#logger)

>   - [`createRequestLogMessage`](#createRequestLogMessage)

Reviewer 2: I'm not particularly fond of this name. I think something like `customRequestLog` would be better.

Author: > I'm not particularly fond of this name. I think something like `customRequestLog` would be better.

No problem I like that too!  I'll make sure to edit the variable name.

  const fastify = require('fastify')({logger: customLogger});
  ```

### `createRequestLogMessage`
<a id="create-request-log-message"></a>

+ Default: `true`

> When logging is enabled via `disableRequestLogging` parameter, users can 
> implement the `createRequestLogMessage` function to customize their log management 
> system.  As a result, this setting allows users to produce a customizable log message 
> catered to their projects' needs, circumventing the default format. 

Reviewer 2: ```suggestion
When request logging is enabled, request logs can be customized by
supplying a `createRequestLogMessage(request, message)` function.
The function must be synchronous, and must not throw an error.
```

Author: Thank you @jsumners, I'll make sure incorporate this.

Example:

```js
  if (disableRequestLogging === false) {
        if (typeof createRequestLogMessage === 'function') {
          childLogger.info({ req: request }, createRequestLogMessage(req, 'customMessage'))
        } else {
          childLogger.info({ req: request }, 'incoming request')
        }

>       }

Reviewer 2: The indentation is not at all consistent in this block. I am also still unclear why this "example" is showing what looks like Fastify's internal implementation of the feature. It should be showing how to use it as user of Fastify.

Author: > The indentation is not at all consistent in this block.

Hi @jsumners, no problem, I'll make sure to re-adjust the indentation.

Author: >  I am also still unclear why this "example" is showing what looks like Fastify's internal implementation of the feature. It should be showing how to use it as user of Fastify.

Ok sounds good, I'll make sure to change that so it's from the user's perspective.

  const fastify = require('fastify')({logger: customLogger});
  ```

### `createRequestLogMessage`
<a id="create-request-log-message"></a>

+ Default: `false`

> When logging is permitted, the user can fully customize their 

Reviewer: Text is not matching the described functionality. default is not false, but basically `v => v`

Author: Ok I'll be sure to correct that thanks.

log settings to suit their needs without relying on the default settings.  
As a result, the user is no longer limited to the default layout.
Please view the following example:

```js
  const customMessage = 'incoming request'

  if (disableRequestLogging === false) {
  childLogger.info({ req: request }, createRequestLogMessage(customMessage))
  } else { childLogger.info({ req: request }, 'incoming request') }

> ```

Author: No problem I will make to revise the documentation.  Is there a specific format you would recommend I follow?

As for the documentation, the message I want to convey is that the user will have the ability to customize the log management system if disableRequestLogging is enabled.  The user will be able to customize this feature to fit their objective.  As for the code, this was a mistake that I should've corrected.  Originally I had it set up that way but then I was encountering an error so I had to change it.

  }
})
```

> 
> 
> 

Reviewer: ```suggestion
```

Author: Thank you I will be sure to remove it.

> 

Author: Ok sounds good will do.
